logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/ProtoContentMapper.kt

```diff
@@ -492,6 +544,40 @@ class ProtoContentMapperImpl(
         )
     }
 
+    private fun unpackComposite(protoContent: GenericMessage.Content.Composite): MessageContent.Composite {
+        val text = protoContent.value.items.firstNotNullOfOrNull { item ->
+            item.text
+        }?.let { text ->
+            MessageContent.Text(
+                value = text.content,
+                mentions = text.mentions.mapNotNull { messageMentionMapper.fromProtoToModel(it) },
+                quotedMessageReference = text.quote?.let { quote ->
+                    MessageContent.QuoteReference(
+                        quotedMessageId = quote.quotedMessageId,
+                        quotedMessageSha256 = quote.quotedMessageSha256?.array,
+                        isVerified = false
+                    )
+                },
+                quotedMessageDetails = null
+            )
+        }
+
+        val buttonList = protoContent.value.items.mapNotNull {
+            it.button?.let { button ->
+                MessageContent.Composite.Button(
+                    text = button.text,
+                    id = button.id,
+                    isSelected = false
+                )
+            }
+        }
+
```

Comment:
how about extracting this stuff to seperate functions and inlining their result within 
```

 return MessageContent.Composite(
            textContent = text,
            buttonList = buttonList
        )
```

Reply:
sure